server/controllers/nlp.js

```diff
@@ -17,19 +21,176 @@ let functions = {
                 type: 'histogram'
             }]
         ))
-    }
+    },
+
+    plotLineChartOfColumn: function (dataset, parameters, callback) {
+        let column = parameters[0];
+
+        // SELECT <column> FROM <dataset>
+        bookshelf.Model.extend({tableName: dataset}).fetchAll({columns: [column]}).then(data => callback(
+            [{
+                // map [{<columnName>: <columnValue>}, ...] to [<columnValue>, ...]
+                x: data.map((value) => value.attributes[column]),
+                type: 'scatter'
+            }]
+        ))
+    },
+    plotScatterOfTwoColumns: function (dataset, parameters, callback) {
+        let column1 = parameters[0];
+        let column2 = parameters[1];
+
+        // SELECT <column> FROM <dataset>
+        bookshelf.Model.extend({tableName: dataset}).fetchAll({columns: [column1, column2]}).then(data => {
+            callback(
+                [{
+                    // map [{<columnName>: <columnValue>}, ...] to [<columnValue>, ...]
+                    x: data.map((value) => value.attributes[column1]),
+                    y: data.map((value) => value.attributes[column2]),
+                    mode: 'markers',
+                    type: 'scatter'
+                }]
+            )
+        });
+    },
+    plotPieChartOfColumn: function (dataset, parameters, callback) {
+        let column = parameters[0];
+        // SELECT <column> FROM <dataset>
+        bookshelf.Model.extend({tableName: dataset}).fetchAll({columns: [column]}).then(data => {
+            var entries = data.map((value) => value.attributes[column]);
+            var values = [];
+            var labels = [];
+            _.forEach(entries, function (entry) {
+                if(!_.includes(labels, entry)){
+                    labels.push(entry);
+                    values.push(1);
+                } else {
+                    var index = labels.indexOf(entry);
+                    values[index]++;
+                }
+            })
+            callback(
+                [{
+                    // map [{<columnName>: <columnValue>}, ...] to [<columnValue>, ...
+                    values: values,
+                    labels: labels,
+                    type: 'pie'
+                }]
+            )
+        })
+
+
+    },
 };
 
-function findCommand(dataset, input) {
-    // TODO replace with NLP logic
-    return {
-        function: 'plotHistogramOfColumn',
-        parameters: [
-            'age'
-        ]
+
+function extractOperation(state) {
+    let possibleOperations = ["plot", "make", "draw", "select"];
+    var currentToken = state.tokens[state.currentToken];
+
+    var result = _.includes(possibleOperations, currentToken);
```

Comment:
please rename this variable in `tokenMatched` in this and all similiar functions